client/src/main/java/org/spine3/base/Stringifiers.java

```diff
@@ -139,38 +146,63 @@ private static String convert(Message message) {
 
     private static String messageWithMultipleFieldsToString(MessageOrBuilder message) {
         String result = shortDebugString(message);
-        result = PATTERN_COLON_SPACE.matcher(result).replaceAll(EQUAL_SIGN);
+        result = PATTERN_COLON_SPACE.matcher(result)
+                                    .replaceAll(EQUAL_SIGN);
         return result;
     }
 
-    static class TimestampIdStringifer implements Stringifier<Timestamp> {
+    protected static class TimestampIdStringifer extends Stringifier<Timestamp, String> {
+
         @Override
-        public String apply(@Nullable Timestamp timestamp) {
-            if (timestamp == null) {
-                return NULL_ID;
-            }
+        protected String doForward(Timestamp timestamp) {
             final String result = toIdString(timestamp);
             return result;
         }
-    }
 
-    static class EventIdStringifier implements Stringifier<EventId> {
         @Override
-        public String apply(@Nullable EventId eventId) {
-            if (eventId == null) {
-                return NULL_ID;
+        @SuppressWarnings("ThrowInsideCatchBlockWhichIgnoresCaughtException")
+        // It is OK, because all necessary information about exception
+        // is passed to the {@code ConversionError} instance.
+        protected Timestamp doBackward(String s) {
+            try {
+                return Timestamps.parse(s);
+            } catch (ParseException e) {
+                final ConversionError conversionError = new ConversionError(e.getMessage(),
+                                                                            e.getErrorOffset());
+                throw new IllegalArgumentException(conversionError);
```

Comment:
I understand your objections.
But I can't throw `ConversionError` explicitly, as it is checked exception and going on inside the overridden `doBackward` Guava method.  So the possible decision is catch all `RuntimeException`, check it with `instanceof` and rethrow.